### PR TITLE
feat(#1226): Check `eo-maven-plugin` version during caching 

### PR DESCRIPTION
**Pull request by volodya-lombrozo**
Added `CacheVersion` class that encapsulates the logic of choosing whether `eo-maven-plugin` version is cacheable or not.

Closes: #1226

### CODE REVIEW

**Comment by volodya-lombrozo** [comment]
@mximp What do you think if we replace `Footprints` [implementation](https://github.com/objectionary/eo/pull/1276) with the new `CacheVersion` class instead?

File: eo-maven-plugin/src/main/java/org/eolang/maven/footprint/CacheVersion.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+ */
+package org.eolang.maven.footprint;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * Cache version.
```
Comment (mximp) [inline eo-maven-plugin/src/main/java/org/eolang/maven/footprint/CacheVersion.java:32]: @volodya-lombrozo Can you please elaborate the description. It's not quite clear what cache is meant. Maybe add what format is represented?

File: eo-maven-plugin/src/main/java/org/eolang/maven/footprint/CacheVersion.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+ */
+final class CacheVersion {
+
+    /**
+     * Not cacheable versions.
+     */
+    private static final String[] NOT_CACHEABLE = {"0.0.0", "SNAPSHOT"};
+
+    /**
+     * Version of the eo-maven-plugin.
```
Comment (mximp) [inline eo-maven-plugin/src/main/java/org/eolang/maven/footprint/CacheVersion.java:44]: @volodya-lombrozo how is this related to `eo-maven-plugin`? Is some format implied?

File: eo-maven-plugin/src/main/java/org/eolang/maven/footprint/CacheVersion.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+     */
+    private final String version;
+
+    /**
+     * Hash of the object.
```
Comment (mximp) [inline eo-maven-plugin/src/main/java/org/eolang/maven/footprint/CacheVersion.java:49]: @volodya-lombrozo is this for single object?

File: eo-maven-plugin/src/main/java/org/eolang/maven/footprint/CacheVersion.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            && !this.hash.isEmpty()
+            && (Objects.isNull(this.version)
+            || Arrays.stream(CacheVersion.NOT_CACHEABLE).noneMatch(this.version::contains));
+    }
+
+    /**
+     * Get cache path.
+     * @return Path.
+     */
+    Path cache() {
```
Comment (mximp) [inline eo-maven-plugin/src/main/java/org/eolang/maven/footprint/CacheVersion.java:78]: @volodya-lombrozo I believe better name would be `path()`. `version.path()` seems to be more natural.

**Comment by volodya-lombrozo** [comment]
@mximp I've renamed `cache()` -> `path()` and added more informative description. Could you check again, please? Maybe we can improve the description even more?

**Comment by mximp** [comment]
@volodya-lombrozo I still believe that referring to `eo-maven-plugin` is misleading. Can't this class be used for any other version? I see the description might be as follows:
```java
/**
  * A version consisting of two parts: 1) base which is maven-style version string
  * and 2) hash string containing git hash. The two parts define the cacheable location be method {@link #path}
  * If base is empty or contains '0.0.0' or 'SNAPSHOT' the version is considered non-cacheable.
*/
```

**Comment by volodya-lombrozo** [comment]
> @volodya-lombrozo I still believe that referring to `eo-maven-plugin` is misleading. Can't this class be used for any other version? I see the description might be as follows:
> 
> ```java
> /**
>   * A version consisting of two parts: 1) base, which is maven-style version string
>   * and 2) hash string, containing git hash. The two parts define the cacheable location by method {@link #path}
>   * If base is empty or contains '0.0.0' or 'SNAPSHOT' the version is considered non-cacheable.
> */
> ```

@mximp Thanks! Updated. Could you have a look, again, please?

**Review by mximp — APPROVED** [review]

**Comment by volodya-lombrozo** [comment]
@yegor256 Could you merge that changes, please?

**Comment by yegor256** [comment]
@rultor merge
